test/end2end/generated_mocks_test.dart

```diff
@@ -229,6 +260,21 @@ void main() {
       when(baz.returnsTypeVariable()).thenReturn(3);
       baz.returnsTypeVariable();
     });
+
+    test(
+      'access typeVariableField will throw UnsupportedError before '
+      'setting default via setDefaultDummyValue',
+      () => expect(() => baz.typeVariableField, throwsUnsupportedError),
+    );
+
+    test(
```

Comment:
I'd like some more tests for `setDefaultDummyValue`. They could be added here or in another file like `mockito_test.dart`. They can be shaped just like this one, where `setDefaultDummyValue` is set, and mock method is called:

* `setDefaultDummyValue()` without a type argument
* `setDefaultDummyValue<X>()` with `typedef X = int`
* `setDefaultDummyValue<Foo<Bar>>()` (both Foo and Bar being imported)
* Maybe something showing subtypes don't work, so `setDefaultDummyValue<num>(7)` and then `baz.returnsTypeVariable<int>()` I think shouldn't work.